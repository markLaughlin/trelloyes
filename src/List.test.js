import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('My component', () => {
    //all my test cases here
    //the cards array is passed into the List component from the App component. It does not exist independently in List
    //The List test failed because it did not have access to the cards array when it went through the List component
    //Therefore we put cards in the List component in the List test as an empty array, which means it is no longer undefined
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render((<List cards= {[]}/>), div);
        ReactDOM.unmountComponentAtNode(div);
      });

      it('renders correctly; snapshot test', () => {
        const tree = renderer
          .create(<List cards= {[]}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });


   });


   
  
