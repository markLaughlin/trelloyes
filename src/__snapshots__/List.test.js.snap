// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My component renders correctly; snapshot test 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
